sos/plugins/crash_report.py

```diff
@@ -0,0 +1,256 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+import os
+import sys
+import time
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+from sos.utilities import sos_get_command_output, find
+from datetime import datetime
+
+
+class CrashLog(Plugin):
+    """Crash kernel report
+    """
+    plugin_name = "crash_report"
+    profiles = ('system', 'debug')
+
+    dbg_vmlinux_path = '/usr/lib/debug/'
+    vmcore_stnd_path = '/var/crash'
+
+    dump_file_prefix = ''
+    crashed_kernel_name = ''
+    vmlinux_path = ''
+    message_to_log = ''
+    vmcore_file_path = ''
+    crash_script = ""
+    # crash result output file
+    crashed_output_file = '/var/log/os_crash_report.txt'
```

Comment:
Thanks Bryn for quick response.
Yes I do get sos archive path from "get_cmd_output_path".

But If i create any file under my plugin directory inside **sos archive** 
             (/var/tmp/sos.XYZ/sos_commands/crash_report/file_created). 
This file will become part of tarball only if i pass file as an input to below call:
1. add_cmd_output
2. add_copy_spec

Issue i am facing to use above two calls are:
1. add_cmd_output:
-----------------------------
1. Got sos archive path for my plugin using "get_cmd_output_path" 
                                         -- /var/tmp/sos.xx/sos_commands/plugin_name/
2. passed my command and file name(in which i want to save output) to "add_cmd_output" call.
Issue in above call in my plug-in is:
When i run crash command, it provides it's command prompt on which i type command which i want to execute on crash prompt.
function "add_cmd_output" doesn't provide an interface to provide comand as aninput and hence we won't be able to run command on crash prompt and it hangs.
Example:
How do we provide input on crash prompt:
1. run crash command:
     crash -s <PATH OF VMLINUX>   <PATH OF VMCORE FILE>
2. Above command opens a crash command prompt on which we will trigger crash command
    crash> 
3. Enter command:
    crash> sys


1. add_copy_spec:
--------------------------------
Steps followed:
1. Got sos archive path for my plugin using "get_cmd_output_path" 
                                         -- /var/tmp/sos.xx/sos_commands/plugin_name/
2. appended my file string to above path to create file
3. created file.
4. paased my file name as an input to "add_copy_spec" 
Issue here is:
 - It adds my file under tarball with complete location 
                  sosreport.../var/tmp/sos_zyz/sos_commands/crash_report/filename
    Which we of course don't want.
          File should be under sosreport.../sos_commands/crash_report/filename